pkg/auth/service.go

```diff
@@ -959,3 +964,626 @@ func (s *DBAuthService) Authorize(ctx context.Context, req *AuthorizationRequest
+			UserID:                        credentials.UserId,
+		}, nil
+	})
+}
+
+func (a *APIAuthService) ListUserCredentials(ctx context.Context, username string, params *model.PaginationParams) ([]*model.Credential, *model.Paginator, error) {
+	resp, err := a.apiClient.ListUserCredentialsWithResponse(ctx, username, &ListUserCredentialsParams{
+		Prefix: paginationPrefix(params.Prefix),
+		After:  paginationAfter(params.After),
+		Amount: paginationAmount(params.Amount),
+	})
+	if err != nil {
+		return nil, nil, err
+	}
+	if err := expectedStatusCode(resp, 200); err != nil {
+		return nil, nil, err
+	}
+
+	credentials := make([]*model.Credential, len(resp.JSON200.Results))
+
+	for i, r := range resp.JSON200.Results {
+		credentials[i] = &model.Credential{
+			AccessKeyID: r.AccessKeyId,
+			IssuedDate:  time.Unix(r.CreationDate, 0),
+			UserID:      0,
+		}
+	}
+	return credentials, toPagination(resp.JSON200.Pagination), nil
+}
+
+func (a *APIAuthService) AttachPolicyToUser(ctx context.Context, policyDisplayName, username string) error {
+	resp, err := a.apiClient.AttachPolicyToUserWithResponse(ctx, username, policyDisplayName)
+	if err != nil {
+		return err
+	}
+	return expectedStatusCode(resp, 201)
+}
+
+func (a *APIAuthService) DetachPolicyFromUser(ctx context.Context, policyDisplayName, username string) error {
+	resp, err := a.apiClient.DetachPolicyFromUserWithResponse(ctx, policyDisplayName, username)
+	if err != nil {
+		return err
+	}
+	return expectedStatusCode(resp, 204)
+}
+
+func (a *APIAuthService) listUserPolicies(ctx context.Context, username string, params *model.PaginationParams, effective bool) ([]*model.Policy, *model.Paginator, error) {
+	resp, err := a.apiClient.ListUserPoliciesWithResponse(ctx, username, &ListUserPoliciesParams{
+		Prefix:    paginationPrefix(params.Prefix),
+		After:     paginationAfter(params.After),
+		Amount:    paginationAmount(params.Amount),
+		Effective: &effective,
+	})
+	if err != nil {
+		return nil, nil, err
+	}
+	if err := expectedStatusCode(resp, 200); err != nil {
+		return nil, nil, err
+	}
+	policies := make([]*model.Policy, len(resp.JSON200.Results))
+
+	for i, r := range resp.JSON200.Results {
+		policies[i] = serializePolicyToModalPolicy(r)
+	}
+	return policies, toPagination(resp.JSON200.Pagination), nil
+}
+
+func (a *APIAuthService) ListUserPolicies(ctx context.Context, username string, params *model.PaginationParams) ([]*model.Policy, *model.Paginator, error) {
+	return a.listUserPolicies(ctx, username, params, false)
+}
+
+func (a *APIAuthService) ListEffectivePolicies(ctx context.Context, username string, params *model.PaginationParams) ([]*model.Policy, *model.Paginator, error) {
+	if params.Amount == -1 {
+		// read through the cache when requesting the full list
+		policies, err := a.cache.GetUserPolicies(username, func() ([]*model.Policy, error) {
+			policies, _, err := a.listUserPolicies(ctx, username, params, true)
+			return policies, err
```

Comment:
use a different names inside the scope so the reader will not get mixed. or just return the 3 of them and ignore the middle one outside